sense.go

```diff
@@ -0,0 +1,220 @@
+package rpmpack
+
+import (
+	"fmt"
+	"regexp"
+)
+
+type rpmSense uint32
+
+// SenseAny specifies no specific version compare
+// SenseLess specifies less then the specified version
+// SenseGreater specifies greater then the specified version
+// SenseEqual specifies equal to the specified version
+const (
+	SenseAny  rpmSense = 0
+	SenseLess rpmSense = 1 << iota
+	SenseGreater
+	SenseEqual
+)
+
+type relationCategory string
+
+const (
+	RequiresCategory   relationCategory = "requires"
+	ObsoletesCategory  relationCategory = "obsoletes"
+	SuggestsCategory   relationCategory = "suggests"
+	RecommendsCategory relationCategory = "recommends"
+	ConflictsCategory  relationCategory = "conflicts"
+	ProvidesCategory   relationCategory = "provides"
+)
+
+var relationMatch = regexp.MustCompile(`([^=<>\s]*)\s*((?:=|>|<|>=|<=)*)\s*(.*)?`)
+
+// Relation is the structure of rpm sense relationships
+type Relation struct {
+	Name    string
+	Version string
+	Sense   rpmSense
+}
+
+// String return the string representation of the Relation
+func (r *Relation) String() string {
+	return fmt.Sprintf("%s%v%s", r.Name, r.Sense, r.Version)
+}
+
+// GoString return the string representation of the Relation
+func (r *Relation) GoString() string {
+	return r.String()
+}
+
+// Equal compare the equality of two relations
+func (r *Relation) Equal(o *Relation) bool {
+	return r.String() == o.String()
+}
+
+// Relations is a slice of Relation pointers
+type Relations []*Relation
+
+// String return the string representation of the Relations
+func (r *Relations) String() string {
+	var (
+		val   string
+		total = len(*r)
+	)
+
+	for idx, relation := range *r {
+		val += fmt.Sprintf("%s%v%s", relation.Name, relation.Sense, relation.Version)
+		if idx < total-1 {
+			val += ","
+		}
+	}
+
+	return val
+}
+
+// GoString return the string representation of the Relations
+func (r *Relations) GoString() string {
+	return r.String()
+}
```

Comment:
Why do we need the string representations here?
If it's because of testing, you should use cmp.Diff https://godoc.org/github.com/google/go-cmp/cmp#Diff 
Then just do 
```golang
if d:= cmp.Diff(expected, got); d!="" {
  t.Errorf("unexpected diff: %s", d),
}
```

Then get rid of the string functions, they add complication with no clear benefit.

If you need it for the `Equal` method, then just comparing the struct members feels cleaner then constructing a string.

This is true for all of the stringers you have here. I would rather avoid that.